File: core/web/src/main/java/edu/uci/ics/texera/web/response/GenericWebResponse.java
package edu.uci.ics.texera.web.response;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * This is a response bean class, that is required for
 * Jackson to serialize an object of this class to JSON
 * Created by kishore on 10/4/16.
 */
public class GenericWebResponse {
    private int code;
    private String message;
    
    public static GenericWebResponse generateSuccessResponse() {
        return new GenericWebResponse(0, "success");
    }

    public GenericWebResponse() {
        // Default constructor is required for Jackson JSON serialization
    }

    public GenericWebResponse(int code, String message) {
        this.code = code;
        this.message = message;
    }

    @JsonProperty
    public int getCode() {
        return code;
    }

    @JsonProperty
    public String getMessage() {
        return message;
    }
}

File: core/amber/src/main/scala/edu/uci/ics/texera/web/model/response/GenericWebResponse.scala
package edu.uci.ics.texera.web.model.response

import com.fasterxml.jackson.annotation.JsonProperty

// TODO: change the response, get rid of the message wrapper.
object GenericWebResponse {
  def generateSuccessResponse = new GenericWebResponse(0, "success")
}

class GenericWebResponse() // Default constructor is required for Jackson JSON serialization
{
  private var code = 0
  private var message = ""

  def this(code: Int, message: String) {
    this()
    this.code = code
    this.message = message
  }

  @JsonProperty def getCode: Int = code

  @JsonProperty def getMessage: String = message
}
